protocol-units/da/movement/celestia/util/src/ir_blob.rs

```diff
@@ -19,12 +19,33 @@ pub struct InnerSignedBlobV1Data {
 	pub timestamp: u64,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Id(Vec<u8>);
```

Comment:
The dedicated newtype is better, but I hoped it'd be a non-allocating fixed size array as well, so that it implements `Copy`.

I guess this will have to wait until someone figures out a reasonable system of bounds to abstract the hash algorithm. The hash is always of fixed size, if I understand the currently used bounds correctly.

Reply:
Yes, but yes extracting the size is the trick. I had done this once somewhere else. Not going to worry about it here. Feel free to put in an issue.